panda/src/cocoadisplay/cocoaGraphicsWindow.mm

```diff
@@ -568,20 +568,45 @@
   }
 
   if (_properties.has_cursor_filename()) {
-    NSImage *image = load_image(_properties.get_cursor_filename());
+    NSData *image_data = load_image_data(_properties.get_cursor_filename());
+
+    // Read the metadata from the image, which should contain hotspotX and
+    // hotspotY properties.
+    CGImageSourceRef cg_image = CGImageSourceCreateWithData((CFDataRef)image_data, nullptr);
+    NSDictionary *image_props = (NSDictionary *)CGImageSourceCopyPropertiesAtIndex(cg_image, 0, nil);
```

Comment:
This is a copy operation, yet the properties object is not being released.

To be on the safe side, perhaps we can also check that `image_props` isn't null before accessing it.  I've been bitten before by this kind of thing.